api/main.py

```
@@ -0,0 +1,152 @@
+    
+    result = mine_and_process_block(blockchain, mempool, pending_nonce_map)
+    
+    if result[0]:
+        return {"message": "Block mined successfully", "block": result[0].to_dict()}
+    else:
+        raise HTTPException(status_code=400, detail="Failed to mine block")
+
+
+if __name__ == "__main__":
+    import uvicorn
+    uvicorn.run(app, host="0.0.0.0", port=8000)
```

Ruff (0.15.1)

[error] Possible binding to all interfaces

(S104)